app/models/friendship.rb

```diff
@@ -0,0 +1,67 @@
+class Friendship < ActiveRecord::Base
+  extend TokenContext
+  include Tokenizer
+	enum friendship_status: [ :denied, :pending, :accepted]
```

Comment:
It's awesome that you're using an enum here.  I feel like this may also have been useful in your friendships controller.  Not sure though, and I'm too lazy to look back and confirm that...